modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/files_info_query.rb

```diff
@@ -0,0 +1,157 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages::Peripherals::StorageInteraction::Nextcloud
+  class FilesInfoQuery
+    using Storages::Peripherals::ServiceResultRefinements
+
+    FILES_INFO_PATH = 'ocs/v1.php/apps/integration_openproject/filesinfo'.freeze
+
+    def initialize(storage)
+      @uri = URI(storage.host).normalize
+      @oauth_client = storage.oauth_client
+    end
+
+    def call(user:, file_ids: [])
+      if file_ids.nil?
+        return Util.error(:error, 'File IDs can not be nil', file_ids)
+      end
+
+      if file_ids.empty?
+        return ServiceResult.success(result: [])
+      end
+
+      Util.token(user:, oauth_client: @oauth_client) do |token|
+        files_info(file_ids, token).map(&parse_json) >> handle_failure >> create_storage_file_infos
+      end
+    end
+
+    private
+
+    def files_info(file_ids, token)
+      ServiceResult.success(
+        result: RestClient::Request.execute(
+          method: :post,
+          url: Util.join_uri_path(@uri, FILES_INFO_PATH),
+          payload: { fileIds: file_ids }.to_json,
+          headers: {
+            'Authorization' => "Bearer #{token.access_token}",
+            'Accept' => 'application/json',
+            'Content-Type' => 'application/json',
+            'OCS-APIRequest' => true
+          }
+        )
+      ).map(&:body)
+    rescue RestClient::Unauthorized => e
+      Util.error(:not_authorized, 'Outbound request not authorized!', e.response)
+    rescue RestClient::NotFound => e
+      Util.error(:not_found, 'Outbound request destination not found!', e.response)
+    rescue RestClient::ExceptionWithResponse => e
+      Util.error(:error, 'Outbound request failed!', e.response)
+    rescue StandardError
+      Util.error(:error, 'Outbound request failed!')
+    end
+
+    def parse_json
+      ->(response_body) do
+        # rubocop:disable Style/OpenStructUse
+        JSON.parse(response_body, object_class: OpenStruct)
+        # rubocop:enable Style/OpenStructUse
+      end
+    end
+
+    def handle_failure
+      ->(response_object) do
+        if response_object.ocs.meta.status == 'ok'
+          ServiceResult.success(result: response_object)
+        else
+          Util.error(:error, 'Outbound request failed!', response_object)
+        end
+      end
+    end
+
+    # rubocop:disable Metrics/AbcSize
+    def create_storage_file_infos
+      ->(response_object) do
+        ServiceResult.success(
+          result: response_object.ocs.data.to_h.map do |key, value|
```

Comment:
> `to_h.map` seems superfluous, specially since you are iterating over key and value, a simpler call would be `map do |(key, value)|`.

Didn't know that, thx. (was not suggested by the IDE and I don't read all the APIs ;)

> Are we covering for possible null data? Would a call to the more strict Hash() method be better then? This method raises an ArgumentError in case the underlying value is impossible to transform into a hash (think of a number or string).

Good point. Would it be a domain error or a technical one? A domain error I would transpose into a `StorageError`.

Reply:
In this case I'd make it a technical one, as the remote response sent us garbage or changed its behaviour and we didn't react in time.

Reply:
Since that's a `ostruct`, you can call `.each_pair.map` which will avoid the intermediate transformation into a hash.